applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_central_differences_scheme.hpp

```diff
@@ -0,0 +1,922 @@
+            array_1d<double,3>& current_angular_velocity      = i->FastGetSolutionStepValue(ANGULAR_VELOCITY);
+            array_1d<double,3>& current_residual_moment       = i->FastGetSolutionStepValue(MOMENT_RESIDUAL);
+            array_1d<double,3>& current_rotation              = i->FastGetSolutionStepValue(ROTATION);    
+            
+            for (unsigned int j =0; j<3; j++)
+            {
+              
+              middle_angular_velocity[j]      = current_angular_velocity[j] ;
+              current_residual_moment[j]     = 0.0;
+              current_rotation[j] = 0.0;
+            }
+          }
+
+
+
+        }
+      }
+    KRATOS_CATCH("")
+	  }
+
+  /**
+      Performing the update of the solution.
+  */
+  //***************************************************************************
+    virtual void Update(ModelPart& rModelPart,
+          DofsArrayType& rDofSet,
+          TSystemMatrixType& A,
+          TSystemVectorType& Dx,
+          TSystemVectorType& b
+          )
+    {
+      KRATOS_TRY
+      ProcessInfo& rCurrentProcessInfo  = rModelPart.GetProcessInfo();
+      NodesArrayType& rNodes            = rModelPart.Nodes();
+      const double numerical_limit = std::numeric_limits<double>::epsilon();
+      //Step Update
+      mTime.Current   = rCurrentProcessInfo[TIME];  //the first step is time = initial_time ( 0.0) + delta time
+      mTime.Delta     = rCurrentProcessInfo[DELTA_TIME];
+
+      mTime.Middle    = 0.5*(mTime.Previous + mTime.Current);
+
+
+      #ifdef _OPENMP
+        int number_of_threads = omp_get_max_threads();
+      #else
+        int number_of_threads = 1;
+      #endif
+
+      vector<unsigned int> node_partition;
+      OpenMPUtils::CreatePartition(number_of_threads, rNodes.size(), node_partition);
+
+      #pragma omp parallel for
+      for(int k=0; k<number_of_threads; k++)
+      {
+        typename NodesArrayType::iterator i_begin=rNodes.ptr_begin()+node_partition[k];
+        typename NodesArrayType::iterator i_end=rNodes.ptr_begin()+node_partition[k+1];
+
+        for(ModelPart::NodeIterator i=i_begin; i!= i_end; ++i)
+        {
+          //Current step information "N+1" (before step update).
+
+          const double& nodal_mass                    = i->FastGetSolutionStepValue(NODAL_MASS);
+          array_1d<double,3>& current_residual        = i->FastGetSolutionStepValue(FORCE_RESIDUAL);
+
+          array_1d<double,3>& current_velocity        = i->FastGetSolutionStepValue(VELOCITY);
+          array_1d<double,3>& current_displacement    = i->FastGetSolutionStepValue(DISPLACEMENT);
+          array_1d<double,3>& middle_velocity         = i->FastGetSolutionStepValue(MIDDLE_VELOCITY);
+
+          array_1d<double,3>& current_acceleration    = i->FastGetSolutionStepValue(ACCELERATION);
+
+          //Solution of the explicit equation:
+          if (nodal_mass > numerical_limit)  current_acceleration = current_residual/nodal_mass;
```

Comment:
should this throw if nodal mass is small?

Reply:
no i still want to continue the simulation in that case but with zero acc